### PR TITLE
Reset values flag

### PR DESCRIPTION
**Pull request by kkoralsky**
implemented `--reset-values` flag overriding `.helmDefaults.reuseValues=true`

as discussed: https://github.com/helmfile/helmfile/discussions/593

### CODE REVIEW

**Comment by kkoralsky** [comment]
@yxxhero would you mind to take a look at this? We talked about the problem in "discussions" week ago :) Im also open for suggestions if it can be implemented better

**Comment by yxxhero** [comment]
I will review as soon as I can.

**Comment by yxxhero** [comment]
@kkoralsky  when has --reset-vaules and --reuse-values in same. what will happen?

**Comment by yxxhero** [comment]
@kkoralsky we can follow the behavior of helm.

File: pkg/state/state.go [hunk 1]
```diff
@@ -619,7 +619,7 @@ func (st *HelmState) prepareSyncReleases(helm helmexec.Interface, additionalValu
 					flags = append(flags, "--wait-for-jobs")
 				}
 
-				if opts.ReuseValues || st.HelmDefaults.ReuseValues {
+				if !opts.ResetValues && (st.HelmDefaults.ReuseValues || opts.ReuseValues) {
 					flags = append(flags, "--reuse-values")
 				} else {
 					flags = append(flags, "--reset-values")
```
Comment (yxxhero) [inline pkg/state/state.go:625]: we can reuse the logic. it can be a function.

File: pkg/state/state.go [hunk 2]
```diff
@@ -1706,7 +1707,7 @@ func (st *HelmState) commonDiffFlags(detailedExitCode bool, includeTests bool, s
 		flags = append(flags, "--output", opt.Output)
 	}
 
-	if opt.ReuseValues || st.HelmDefaults.ReuseValues {
+	if !opt.ResetValues && (opt.ReuseValues || st.HelmDefaults.ReuseValues) {
```
Comment (yxxhero) [inline pkg/state/state.go:1710]: here is a repeat.

File: pkg/state/state_test.go [hunk 3]
```diff
@@ -2789,6 +2799,16 @@ func TestPrepareSyncReleases(t *testing.T) {
...
+		{
+			name:  "force-reset-values",
+			flags: []string{"--reset-values"},
+			syncOptions: &SyncOpts{
+				ResetValues: true,
+			},
+			helmDefaults: &HelmSpec{
+				ReuseValues: true,
+			},
+		},
```
Comment (yxxhero) [inline pkg/state/state_test.go:2811]: add a test case for --reset-values and --reuse-values in the same.

File: pkg/config/sync.go [hunk 4]
```diff
@@ -110,7 +112,13 @@ func (t *SyncImpl) WaitForJobs() bool {
 
 // ReuseValues returns the ReuseValues.
 func (t *SyncImpl) ReuseValues() bool {
-	return t.SyncOptions.ReuseValues
+	if !t.ResetValues() {
+		return t.SyncOptions.ReuseValues
+	}
+	return false
+}
```
Comment (yxxhero) [inline pkg/config/sync.go:119]: could you explain the logic.Thanks very much.

File: pkg/state/state.go (hunk 1 above)
Comment (kkoralsky) [reply pkg/state/state.go]: ok

File: pkg/state/state.go (hunk 2 above)
Comment (kkoralsky) [reply pkg/state/state.go]: ok

File: pkg/state/state_test.go (hunk 3 above)
Comment (kkoralsky) [reply pkg/state/state_test.go:2831]: added

**Review by kkoralsky — COMMENTED** [review]
If both --reset-values and --reuse-values are given, we use *reset* "values control mode" - this is also how helm beahaves

**Comment by yxxhero** [comment]
@kkoralsky please fix DCO issue.

**Comment by yxxhero** [comment]
@mumoshu I think it's ok. do you have some suggestions?

**Comment by yxxhero** [comment]
@kkoralsky BTW. please update the docs.

File: pkg/app/config.go [hunk 5]
```diff
@@ -258,4 +258,5 @@ type InitConfigProvider interface {
 // when enable reuse-values, reuse the last release's values and merge in any overrides values.
```
Comment (yxxhero) [inline pkg/app/config.go:258]: add more comments.

Comment (kkoralsky) [reply pkg/app/config.go]: Ive just made this comment more adequate now

File: pkg/state/state.go [hunk 6]
```diff
@@ -2633,6 +2627,16 @@ func (st *HelmState) chartVersionFlags(release *ReleaseSpec) []string {
...
 
+func (st *HelmState) appendValuesControlModeFlag(flags []string, reuseValues bool, resetValues bool) []string {
+	if !resetValues && (st.HelmDefaults.ReuseValues || reuseValues) {
+		flags = append(flags, "--reuse-values")
+	} else {
+		flags = append(flags, "--reset-values")
+	}
+
+	return flags
+}
```
Comment (yxxhero) [inline pkg/state/state.go:2638]: please add unittest for the func. @kkoralsky

**Comment by yxxhero** [comment]
rest of is LGTM. 
@mumoshu WDYT?

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks a lot for your contribution @kkoralsky and review @yxxhero ☺️

**Pull request merged**